drivers/rn2xx3/rn2xx3.c

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_rn2xx3
+ * @{
+ *
+ * @file
+ * @brief       Driver implementation for the RN2483/RN2903 devices
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <errno.h>
+
+#include "assert.h"
+#include "xtimer.h"
+#include "rn2xx3_params.h"
+#include "rn2xx3.h"
+#include "rn2xx3_internal.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/**
+ * @brief   Delay when resetting the device, 10ms
+ */
+#define RESET_DELAY                 (10UL * US_PER_MS)
+
+/*
+ * Interrupt callbacks
+ */
+static void _rx_cb(void *arg, uint8_t c)
+{
+    rn2xx3_t *dev = (rn2xx3_t *)arg;
+    netdev_t* netdev = (netdev_t *)dev;
+
+    /* The device closes the response with \r\n when done */
+    if (c == '\r') {
+        return;
+    }
+    else if (c == '\n') { /* response end delimiter */
+        dev->resp_buf[dev->resp_size++] = '\0';
```

Comment:
Here I trust the module answer which should be deterministic. The size of the response buffer is large enough to handle all of them normally. I added a check at the beginning of the callback, just in case.